src/qibolab/platforms/abstract.py

```diff
@@ -152,6 +163,135 @@ def reload_settings(self):
             else:
                 self.qubits[q] = Qubit(q, **settings["characterization"]["single_qubit"][q])
 
+    def dump(self, path: Path):
+        with open(path, "w") as file:
+            yaml.dump(self.settings, file, sort_keys=False, indent=4, default_flow_style=None)
+
+    def update(self, updates: dict):
```

Comment:
What is the use case for this function? What parameters does the `updates` dict contain? I am not sure why we need to check all parameter names one by one to find which one to update.

To me it would seem much simpler if in the calibration routine or fitting, where you know what you are updating and you also have access to the platform, you just do for example
```py
platform.qubits[qubit].qubit_frequency = frequency
```
and this is sufficient to register the updated value everywhere needed in qibolab (platform, instruments, runcards when you dump, etc.).

Reply:
The idea was to delegate the update of the runcard params to the platform, keeping qibocal as much agnostic as we can. 
Some fitted parameters are producing different actions to be taken depending on the platform (f.e. threshodls and iq_angles in qblox). So, if we pass the fitted data from qibocal to the platform, we can determine wich actions must be taken depending on the platform.

Reply:
"updates" contains the multiple parameters obtained after fitting

Reply:
Thanks for the comment @stavros11.
During the presentation I didn't mention how to runcard will be update using the new layout.
This encoded in the `Results` object of each routine.
```py
@dataclass
class T1Results(Results):
    """T1 outputs."""

    t1: Dict[List[Tuple], str] = field(metadata=dict(update="t1"))
    """T1 for each qubit."""
    fitted_parameters: Dict[List[Tuple], List]
    """Raw fitting output."""

```
In this case the dictionary produced is the following:
```
{'t1': {0: 8014.838170101057}}
```
Basically in the `Results` class we specify that we want to generate a dictionary with a key "t1", in this way we can disentangle the name of the variable from the key in the platform runcard.
The update method takes this dictionary and update the platform object accordingly.

Reply:
> To me it would seem much simpler if in the calibration routine or fitting, where you know what you are updating and you also have access to the platform, you just do for example
> 
> ```python
> platform.qubits[qubit].qubit_frequency = frequency
> ```
> 
> and this is sufficient to register the updated value everywhere needed in qibolab (platform, instruments, runcards when you dump, etc.).

I think that this is not correct given that currently we have a redundancy in the runcard since we need to update both the qubits attributes and the native gates.

Moreover, it seems that some instruments needs to be re-updated after each routine.

Reply:
Although once #387 is in main this could solve some problems.